sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/ServiceBusTriggerAttributeBindingProvider.cs

```diff
@@ -53,33 +53,27 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
                 return Task.FromResult<ITriggerBinding>(null);
             }
 
-            string queueName = null;
-            string topicName = null;
-            string subscriptionName = null;
-            string entityPath = null;
+            attribute.Connection = _nameResolver.ResolveWholeString(attribute.Connection);
+            string entityPath;
             EntityType entityType;
-
             if (attribute.QueueName != null)
             {
-                queueName = Resolve(attribute.QueueName);
+                var queueName = _nameResolver.ResolveWholeString(attribute.QueueName);
```

Comment:
In EH I moved all these resolves into the client factory to avoid spreading them around.

Reply:
Yeah I tried that, but the resolved names were needed in too many different places for it to work out here.